ui/lib/utils/hooks/useStakedEth.ts

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react'
+import { Contract, utils } from "ethers";
+import { keccak256 } from 'ethers/lib/utils'
+import { ethereum } from 'thirdweb/chains'
+import useContract from '@/lib/thirdweb/hooks/useContract'
+import {STAKED_ETH_ADDRESS, MOONDAO_TREASURY} from "const/config";
+import StakedEthABI from "const/abis/StakingContract.json";
+import { readContract } from 'thirdweb'
+import { ethers5Adapter } from 'thirdweb/adapters/ethers5'
+import client from 'lib/thirdweb/client'
+
+export default function useStakedEth() {
+  const [stakedEth, setStakedEth] = useState<any>()
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<any>(null)
+  const contract = useContract({
+    address: STAKED_ETH_ADDRESS,
+    abi: StakedEthABI as any,
+    chain: ethereum,
+  })
+  const provider = ethers5Adapter.provider.toEthers({
+    client,
+    chain: ethereum,
+  })
+  const ethersContract = new Contract(
+    STAKED_ETH_ADDRESS,
+    StakedEthABI,  // include at least the Deposit event ABI
+    provider
+  );
+
+  useEffect(() => {
+      async function fetchStakedEth() {
```

Comment:
It's too bad that gnosis doesn't just natively recognize the staked ETH. Especially since what we're using is their native staking service - nice work though